daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/XMLTextInfosetOutputter.scala

```diff
@@ -156,8 +158,11 @@ class XMLTextInfosetOutputter private (writer: java.io.Writer, pretty: Boolean)
 
     if (!isNilled(simple) && simple.hasValue) {
       if (simple.erd.optPrimType.get == NodeInfo.String) {
-        val simpleVal = simple.dataValueAsString
-        if (simple.erd.runtimeProperties.get(XMLTextInfoset.stringAsXml) == "true") {
+        var simpleVal = simple.dataValueAsString
+        if (xmlOutputStyle == XMLOutputStyle.PrettyPrintSafe){
```

Comment:
I think this check and simpleVal change wants to be in the other else block. If `stringAsXML` is true, then we don't care at all about this pretty print stuff since the string will be output as raw XML. I think the logic wants to be something like:

```scala
val simpleVal = simple.dataValueAsString
if (....stringAsXml == "true") {
  writeSringAsXml(simpleVal)
} else {
  val simpleValEscaped = xmlOutputStyle match {
    case XMLOutputStyle.Standard => // Utility.escape stuff
    case XMLOutputStyle.PrettyPrintSafe => // CDATA format stuff
  }
  writer.write(simpleValEscaped)
}
```
This also easily extends to other XMLOutputStyles if we ever add more.

Reply:
This code block look better to me as well.  Even better, the match expression will be guaranteed to throw a NoMatchException if the Java compiler lets a Java caller pass null into xmlOutputStyle.